### PR TITLE
[ticket/15925] Add core events for sync and mcp functions

### PR DESCRIPTION
**Pull request by toxyy**
Add core events to modify topic and forum info when syncing
Also add core events to account for missing variables when using mcp functions
Custom post, topic, or forum columns use these events to account for their data

PHPBB3-15925

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [ ] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15925

### CODE REVIEW

**Comment by toxyy** [comment]
I didn't cancel those commits... this has happened before.  Not sure what's going on.

**Comment by toxyy** [comment]
Is this that wrong end of file formatting again or something?  Is it possible to fix within github?  I saved it in notepad++ like I usually do when I make pull requests, not sure whats up with it now.

**Comment by 3D-I** [comment]
The problem IMO is you oughta put a blank new line between the description and the ticket number, like:

```
[ticket/00000] blah blah blah

Add blah..
Also blah..

PHPBB3-00000
```

See: https://area51.phpbb.com/docs/dev/3.2.x/development/git.html

Therefore you need to amend all of those 5 commit messages using a shell like `git bash` or similar.

See: https://help.github.com/articles/changing-a-commit-message/#amending-older-or-multiple-commit-messages

**Comment by 3D-I** [comment]
Ouch, you are now on the master branch (3.3) :nerd_face:

**Comment by toxyy** [comment]
> Ouch, you are now on the master branch (3.3) 

I don't know if you can tell but I'm not really good at using github... 😆 fixed it.  Not sure how that happened...

Thanks for the tip on how to fix the commit message!  Worked like a charm.

**Comment by 3D-I** [comment]
Just close and re-open this one, should do.

**Comment by 3D-I** [comment]
Ok, let it run.

File: phpBB/includes/mcp/mcp_main.php [hunk 1]
```diff
@@ -1455,6 +1455,26 @@ function mcp_fork_topic($topic_ids)
 						$counter[$row['poster_id']] = 1;
 					}
 				}
+
+				/**
+				* Modify the forked post's sql array before inserted into the database.
```
Comment (3D-I) [inline phpBB/includes/mcp/mcp_main.php]: I am not a native speaker but...

Comment (toxyy) [reply phpBB/includes/mcp/mcp_main.php]: I fixed this in the commit message, I'll fix it here too.

**Comment by toxyy** [comment]
> 
> 
> Dunno, seems like the commit message on `Add core.sync_modify_forum_data ` contains a trailing space in the first line, not sure in this case, but fails just over it.

🤦‍♂️ let's try this again.

**Comment by toxyy** [comment]
!set Event

**Pull request closed without merging**